src/util/Range.java

```diff
@@ -34,4 +34,66 @@ public int max(int dimension) {
         assert dimension < dimensions();
         return this.max[dimension];
     }
+
+    public boolean contains(int x) {
+        if(numberOfDimensions > 1)
+            return false;
+        if(x < min[0] || x > max[0])
+            return false;
+        return true;
+    }
+
+    public boolean contains(int x, int y) {
+        if (numberOfDimensions < 2 || numberOfDimensions > 2)
+            return false;
+
+            if(x < min[0] || x > max[0] || y < min[1] || y > max[1])
+                return false;
+        return true;
+    }
+
+    @Override
+    public boolean equals(Object arg) {
+        if(!(arg instanceof Range))
+            return false;
+
+        Range other = (Range)arg;
+        if(other.dimensions() != numberOfDimensions) {
+            return false;
+        }
+        else {
+            for(int i = 0; i < numberOfDimensions; i++) {
+                if(other.min[i] != min[i] || other.max[i] != max[i])
+                    return false;
+            }
+            return true;
+        }
+    }
+
+    public boolean includes(Range arg) {
+        for(int i = 0; i < numberOfDimensions; i++) {
+            if(arg.min(i) < min[i] || arg.max(i) > max[i])
+                return false;
+        }
+        return true;
+    }
+
+    public boolean overlaps(Range arg) {
+        for(int i = 0; i < numberOfDimensions; i++) {
+            if(arg.max(i) < min[i] || arg.min(i) > max[i])
+                return false;
+        }
+        return true;
+    }
+
+    public boolean abuts(Range arg) {
+        if (overlaps(arg))
+            return false;
+
+        for(int i = 0 ; i < numberOfDimensions; i++)
```

Comment:
Make more uniform use of curly braces for one statement blocks. There's no explicit convention but we'd prefer always put curly braces.